exercises/react/example.cr

```diff
@@ -0,0 +1,71 @@
+alias CellValue = Int32
+
+private class Cell
+  @dependents = [] of ComputeCell
+
+  def initialize(initial : CellValue)
+    @value = initial
+  end
+
+  getter :value
+
+  protected getter :dependents
+  protected def propagate; _propagate; end
```

Comment:
Logically, it is also possible to move `hi_friends` up to `Thing` since then `hi_friends` receiver will always be a `Thing` and it is legal for `Thing` to call the `GoodThing#hi` (It is not legal for `SomeThing` to call `GoodThing#hi`). 

In the context of `React`, I don't want to move `hi_friends` (which is `value=`) up to `Thing` (`Cell`), but what I currently have works so it's fine.